packages/gensx-react/src/hooks/use-gensx.ts

```diff
@@ -0,0 +1,498 @@
+import {
+  GenSXComponentEndEvent,
+  GenSXComponentStartEvent,
+  GenSXEndEvent,
+  GenSXErrorEvent,
+  GenSXEvent,
+  GenSXOutputEvent,
+  GenSXProgressEvent,
+  GenSXStartEvent,
+} from "@gensx/client";
+import { useCallback, useMemo, useRef, useState } from "react";
+
+// Match the Client's RunOptions interface
+export interface GenSXRunOptions {
+  org: string;
+  project: string;
+  environment?: string;
+  inputs?: Record<string, unknown>;
+  format?: "sse" | "ndjson" | "json";
+}
+
+export interface UseGenSXOptions<
+  TInputs = unknown,
+  TOutput = unknown,
+  TEvent extends GenSXProgressEvent = GenSXProgressEvent,
+> {
+  /**
+   * API endpoint URL (your server endpoint that proxies to GenSX)
+   */
+  endpoint: string;
+
+  /**
+   * Workflow name to execute (required now)
+   */
+  workflowName: string;
+
+  /**
+   * Default configuration applied to every run/stream (org, project, env, etc.)
+   * Does NOT include inputs – those are passed per invocation.
+   */
+  defaultConfig?: Omit<Partial<GenSXRunOptions>, "inputs">;
+
+  /**
+   * Optional headers to include in the request
+   */
+  headers?: Record<string, string>;
+
+  /**
+   * Callback fired when workflow starts
+   */
+  onStart?: (message: string) => void;
+
+  /**
+   * Callback fired for progress updates
+   * Can receive either a string message or a parsed JSON object for structured progress events
+   */
+  onProgress?: (message: string | any) => void;
+
+  /**
+   * Callback fired for each output chunk (streaming mode only)
+   */
+  onOutput?: (chunk: string) => void;
+
+  /**
+   * Callback fired when workflow completes
+   */
+  onComplete?: (output: TOutput) => void;
+
+  /**
+   * Callback fired on error
+   */
+  onError?: (error: string) => void;
+
+  /**
+   * Callback fired for any event
+   */
+  onEvent?: (event: TEvent) => void;
+}
+
+// Union type for workflow control events (excludes progress and output)
+export type GenSXWorkflowEvent =
+  | GenSXStartEvent
+  | GenSXComponentStartEvent
+  | GenSXComponentEndEvent
+  | GenSXEndEvent
+  | GenSXErrorEvent;
+
+export interface UseGenSXResult<
+  TInputs = any,
+  TOutput = any,
+  TEvent extends GenSXProgressEvent = GenSXProgressEvent,
+> {
+  /** Whether the workflow is currently running */
+  isLoading: boolean;
+
+  /** Whether the workflow is streaming */
+  isStreaming: boolean;
+
+  /** Any error that occurred */
+  error: string | null;
+
+  /** The final output (collection mode) */
+  output: TOutput | null;
+
+  /** All events received */
+  events: GenSXEvent[];
+
+  /** All events received */
+  workflowEvents: GenSXEvent[];
+
+  /** Progress events */
+  progressEvents: TEvent[];
+
+  /** Output events */
+  outputEvents: GenSXOutputEvent[];
+
+  /** Run workflow in collection mode (returns final output) */
+  run: (inputs: TInputs) => Promise<TOutput | null>;
+
+  /** Run workflow in streaming mode */
+  stream: (inputs: TInputs) => Promise<void>;
```

Comment:
@jmoseley I'm curious if it feels right to you to expose separate methods for stream and run? 

Side note, is `run()` actually `/start`? or is it actually `/run`?

Reply:
I just removed stream and made it so it always "streams" which really isn't any different than run for the user